### ISSUE TITLE
🐛[BUG]: CorrDiff-COSMO produces corrupted output with the pinned PhysicsNeMo version

### ISSUE THREAD

**Issue by gertln** [NVIDIA/earth2studio#1145]
### Version

0.18.0a0

### On which installation method(s) does this occur?

pip

### Describe the issue

The CorrDiffCosmoEra5SDA example (examples/05_data_assimilation/03_corrdiff_cosmo_sda.py) produces blocky artifacts  with the PhysicsNeMo revision currently pinned for the `cosmo` extra (ff4cad2). With the earlier PhysicsNeMo (before the RoPE rewrite) the results are correct.

To reproduce install E2S per the official instructions (which pin nvidia-physicsnemo@ff4cad2) and run the example (model hf://nvidia/corrdiff-cosmo-era5@44064f30, INIT_TIME 2024-01-26 00Z).

Expected: smooth analysis, held-out 10 m wind RMSE ~2.1 m/s 
Actual: the guided posterior is has blocky artefacts; held-out RMSE 4.9 -> 6.46

A/B (identical code / checkpoint / data, only PhysicsNeMo changed):
- pre-rewrite (f8d5d2f7): held-out RMSE 4.905 -> 2.147
- pinned (ff4cad2):       held-out RMSE 4.905 -> 6.458
(Good vs broken plots attached.)

<img width="2627" height="778" alt="Image" src="https://github.com/user-attachments/assets/9a375a50-5194-4d9e-bc1f-9c20e837bb82" />
<img width="2640" height="778" alt="Image" src="https://github.com/user-attachments/assets/3f5a334d-249a-4fe4-9d43-3e4d401af46d" />

#Root cause
PhysicsNeMo #1778 reimplemented the RoPE table module (physicsnemo/nn/module/rope.py), so it emits different cos/sin tables for the same grid. The CorrDiff-COSMO DiT has no learned positional parameters (position enters only via RoPE), so the checkpoint is coupled to the RoPE it was trained with. 
E2S #1099 bumped the shared PhysicsNeMo pin (cosmo/stormcast-conus/dlesym) from 94b3551 to ff4cad2, so cosmo now runs against the new RoPE.

**Comment by pzharrington** [comment]
Agentic review/bisection results:

Did a surgical CPU A/B of the two physicsnemo revs (worktrees at `f8d5d2f7` and `94b3551`, components exercised in isolation — no natten, no checkpoint). Two findings:

- **RoPE is not the cause.** `build_axial_rope_cos_sin_2d` and `apply_rotary_pos_emb` are byte-identical across the revs, and the tables come out bit-identical (`maxabsdiff = 0.0`) for several grids *and* after a `_rebind_latent`-style regrid driven through each rev's own API. #1778 only changed table ownership (per-block buffers → one `RotaryEmbedding2DTables` on the DiT); the buffers were `persistent=False` before and after, so nothing about the checkpoint coupling changed. The **unguided** `sample()` loop is also bit-identical, which clears the scheduler/preconditioner/solver path (#1905) too.

- **The real cause is #1757 (tensor-valued DPS params).** `DataConsistencyDPSGuidance.__call__` moved `rho` *inside* the gradient: old was `-grad(loss.sum(), x) / (2*variance)`, new is `-grad((rho*loss).sum(), x)`. Algebraically identical for a scalar `std_y` — but the SDA passes a **field** (`sda_corrdiff_cosmo_era5.py:454`) with `obs_std/s` at observed cells and a `1.0` filler elsewhere. Old: that filler divides the assembled gradient at every *unobserved* location, damping guidance as it spreads off-station through the Jacobian. New: the filler is (correctly) masked out before backprop. Measured guidance ratio new/old is exactly `1.0` at observation cells and exactly `4.0` between them — i.e. `(1.0/0.5)²`, filler variance over obs variance. That factor is structural, independent of the Jacobian, so it carries to the real DiT. Guided sampling differs by ~110–190%.

Two notes on the timeline, since they change where the fix goes:

- **E2S #1099 can't be the trigger.** Both #1778 and #1757 are ancestors of `bf0ad4f4`, the pin in effect when the SDA model itself landed (`49636bf3`, 2026-08-20). `git diff bf0ad4f4 94b3551` is empty over `diffusion/guidance`, `nn/module/rope.py`, `dit_layers.py`, `models/dit`, and `94b3551..ff4cad2` touches nothing under `nn/`, `models/dit`, `diffusion/` or `core/` at all. Every pin this model has ever had behaves identically here.

- **The old behavior was the accidental one.** `f8d5d2f7` predates the SDA by two months and was never an E2S pin; the incidental 4× off-station damping there just happened to flatter the result. The new semantics are correct DPS: the unit filler in `std_y` is fully inert under them (verified — varying it from 1e-3 to 1e6 changes the guidance by exactly 0.0), which is exactly what it was written to be. Under the old code it was *not* inert, because the division happened after the gradient and so was indexed by position in `x` rather than by observation. So the fix is retuning guidance strength on the E2S side (`sda_std_obs` / `sda_gamma`), not a physicsnemo pin revert and not a change to the filler.


Unrelated nit spotted: `_as_broadcastable` calls `torch.as_tensor(x)` with no dtype, so a Python-float `std_y`/`gamma` becomes float32 and is then upcast (`5e-5` → `4.999999873689376e-05`), silently capping the guidance weights at float32 in an otherwise float64 sampler.
